hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java

```diff
@@ -342,16 +342,17 @@ public String getIdentityColumn() {
 		}
 
 		final String[] propertyNames = getIdentifierPropertyNames();
-		List<String> columns = new ArrayList<>();
+		ArrayList<String> columns = new ArrayList<>();
```

Comment:
You could pre-size it like

`ArrayList<String> columns = new ArrayList<>( propertyNames.length );`